File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        def dijgcd(num1,num2):
            if num1 == num2:
                return num1
            if num1 >  num2:
                return dijgcd(num1-num2,num2)
            return dijgcd(num1,num2-num1)
        return dijgcd(min(nums),max(nums))

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        int max = 0;
        int min = 1001;
        for (int i = 0; i < nums.length; i++){
            max = Math.max(max,nums[i]);
            min = Math.min(min,nums[i]);
        }
        return dijalgo(min,max);
    }
    int dijalgo(int a,int b){
        if (a == b){
            return a;
        }
        if (a>b){
            return dijalgo(a-b,b);
        }
        return dijalgo(a,b-a);
    }
}